### ISSUE TITLE
Train all together or individually? How to train spatial subnetwork alone?

### ISSUE THREAD

**Issue by aviziskind** [doubledaibo/drnet_cvpr2017#8]
In the DRNet paper, it says "The appearance module is initialized with a model pre-trained on ImageNet, while the spatial module and the DR-Net are initialized randomly. After initialization, the entire network is jointly optimized using SGD."

But the github README describes several steps to training the network.
1) train A in isolation
2) train S in isolation
3) train A + S in isolation, with weights initialized from previous steps
4) train A + S + D jointly, with weights initialized from previous steps

Is there a significant difference in performance between training the network all together, vs, training in 4 steps?  Also, for training A in isolation, I assume one would train it to output the object classes/labels. But how would one go about training S in isolation? Do you train it to predict the predicate label based solely on the dual spatial masks?

**Comment by doubledaibo** [comment]
Hi @aviziskind,
   I didn't compare training in 4 steps vs training altogether from initialization. 
   Training S in isolation is what you mean, by predicting the predicate label based solely on the dual spatial masks.

**Comment by aviziskind** [comment]
Thanks for clarifying. Would you be willing to share your solver.prototxt files you used for the various training steps?

**Comment by doubledaibo** [comment]
nothing magical, using default or commonly used choices for hyperparameters you could get satisfying results. I think not sharing it is better, because you may think there are some special considerations. but actually, I didn't put much effort on tuning hyperparameters.
